### PR TITLE
Convert IntelliJ plugin to Kotlin

### PR DESCRIPTION
**Pull request by rock3r**
> [!NOTE]
> This requires #501 to be merged first

This PR is the basis for future improvements of the plugin. In this PR, the plugin code is converted to Kotlin, the configurable is recreated with the Kotlin UI DSL, and the goal is to maintain the existing behaviours.

The only small change is that the style dropdown in the configurable is disabled when ktfmt is disabled.

### CODE REVIEW

**Review by hick209 — CHANGES_REQUESTED** [review]
Overall this LGTM, but please mark the files as a move so we don't lose blame history.

**Comment by rock3r** [comment]
Not sure how to do that? I didn't know you could override what diff thinks happened

**Comment by rock3r** [comment]
@hick209 please advise so I can look into it tomorrow

**Comment by rock3r** [comment]
@hick209 I tried looking into this but I did not find a way to force Git to consider files as moved. I had kept the actual conversion of the files to Kotlin to a dedicated commit, too, but that did not help. Also tried re-doing the conversion in another branch by committing the moving first, and the conversion second, but the conversion still results in pairs of adding/deleting

**Comment by hick209** [comment]
Yeah I looked it up as well and noticed this is not really doable in git :-/ 

I'm used to working with Mercurial, which allows this with the following command:
```
hg mv FROM TO --after
```

No worries, I can try to do that internally once we pull your changes. I have to wait until we merge the other 2 PRs for this though.

Thanks a ton for you contributions!

File: ktfmt_idea_plugin/build.gradle.kts [hunk 1]
```diff
@@ -15,48 +17,54 @@
...
+        untilBuild = provider { null }
+    }
+
+    publishing { token = System.getenv("JETBRAINS_MARKETPLACE_TOKEN") }
+
+    pluginVerification { ides { recommended() } }
 }
 
-spotless { java { googleJavaFormat("1.22.0") } }
+spotless { java { googleJavaFormat(libs.versions.googleJavaFormat.get()) } }
```
Comment (hick209) [inline ktfmt_idea_plugin/build.gradle.kts:64]: Now that we're on Kotlin, I guess we could even remove this

Comment (rock3r) [reply ktfmt_idea_plugin/build.gradle.kts:64]: True, will do. Btw, I formatted the Kotlin code with ktfmt, using the Meta style IIRC. Is that the correct one?

**Pull request closed without merging**